src/main/java/it/geosolutions/geoserver/rest/GeoServerRESTPublisher.java

```diff
@@ -3283,8 +3283,12 @@ private boolean recalculateBBox(StoreType type, String xmlElementName, String wo
 //        GSWorkspaceEncoder wsenc = new GSWorkspaceEncoder(workspace);
         
 //        String body = wsenc.toString();
-        String body = "<" + xmlElementName +"><name>" + layerName + "</name>" + 
-                "<enabled>" + enabled + "</enabled></" + xmlElementName + ">";
+//        String body = "<" + xmlElementName +"><name>" + layerName + "</name>" + 
+//                "<enabled>" + enabled + "</enabled></" + xmlElementName + ">";
+        renc.remove(GSResourceEncoder.KEYWORDS);
+        renc.remove(GSResourceEncoder.METADATA);
+        renc.remove(GSResourceEncoder.METADATALINKS);
+        String body = renc.toString();
```

Comment:
When I switched to using GS*Encoder classes, I noted that several extra elements were added to the resulting XML documents. These extra elements could overwrite existing keywords, metadata, and metadatalinks.  While I can remove a hard-coded list of extra elements now, the same kind of problem can occur in the future if new fields are added to the encoders without updating this class. I opted to revert to simple String construction instead of GS*Encoder classes to ensure that this functionality would continue to be stable in the future.